### PR TITLE
[Platform/Ragile] Support RA-B6510-48v8c platform

### PR DESCRIPTION
**Pull request by pettershao-ragilenetworks**
What I did it
Add new platform x86_64-ragile_ra-b6510-48v8c-r0 (Trident 3)
ASIC Vendor: Broadcom
Switch ASIC: Trident 3
Port Config: 48x25G+8x100G

-How I did it
Provide device and platform related files.

-How to verify it
show platform fan
show platform ssdhealth
show platform psustatus
show platform summary
show platform syseeprom
show platform temperature
show interface status

### CODE REVIEW

**Comment by pettershao-ragilenetworks** [comment]
@jleveque help review this new platform, thanks!

File: platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py [hunk 1]
```diff
@@ -0,0 +1,991 @@
+#!/usr/bin/env python3
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py:1]: Why a bespoke fan control program? Why not utilize thermalctld?

File: platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/chassis.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/chassis.py:1]: File is not executable. Please remove shebang.

File: platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/eeprom.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/eeprom.py:1]: File is not executable. Please remove shebang.

File: platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/fan.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/fan.py:1]: File is not executable. Please remove shebang.

File: platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/platform.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/platform.py:1]: File is not executable. Please remove shebang.

File: platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/psu.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+#
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/psu.py:2]: File is not executable. Please remove shebang.

File: platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/sfp.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/sfp.py:1]: File is not executable. Please remove shebang.

File: platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/thermal.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/thermal.py:1]: File is not executable. Please remove shebang.

File: platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/watchdog.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/ra-b6510-48v8c/sonic_platform/watchdog.py:1]: File is not executable. Please remove shebang.

**Review by jleveque — CHANGES_REQUESTED** [review]
As comments.

Also the old platform 1.0 API plugins (in the `plugins/` directory) are deprecated in favor of the new platform API 2.0 (the `sonic_platform` package. Suggest removing the plugins unless you have a good reason to keep them.

File: platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py (hunk 1 above)
Comment (pettershao-ragilenetworks) [reply platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py:1]: The reason is that this file conclude our own defined speed control strategy:
1. We want anti-shake mechanism
2. There is a monitoring MAC temperature, there is a restart mechanism
3. Others

**Comment by pettershao-ragilenetworks** [comment]
@jleveque reply for fancontrol.py, and finished fixing all other files, please help forward!

File: platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py [hunk 10]
```diff
@@ -0,0 +1,991 @@
...
+                self.board_moni_msg()  # re-read
+                if self.check_temp_crit() == True:
+                    logger.crit(
+                        "%%DEV_MONITOR-TEMP:The temperature of device is over critical value.",
+                    )
+                    self.set_fan_max_speed()  # fan full speed
+                    self.critnum += 1  # anti-shake
+                    if self.critnum >= MONITOR_CONST.CRITICAL_NUM:
+                        os.system("reboot")
+                    logger.debug(DEBUG_FANCONTROL, "crit次数:%d" % self.critnum)
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py:914]: Please use ASCII characters only. Please do so throughout all files.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py (hunk 1 above)
Comment (jleveque) [reply platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py:1]: Understood. However, these are enhancements that would be nice-to-haves in SONiC as a whole in the future, and vendors can implement them if they support them. Please feel free to create a HLD and open a PR in https://github.com/Azure/SONiC/pulls to propose adding features to thermalctld or any other SONiC application.

File: platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py (hunk 10 above)
Comment (pettershao-ragilenetworks) [reply platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py]: done!

File: platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py (hunk 1 above)
Comment (pettershao-ragilenetworks) [reply platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py:1]: OK, glad to hear that. We will plan to submit a HLD and make contribution to making these features common to SONiC next stage.

**Review by jleveque — APPROVED** [review]

**Pull request merged**